### PR TITLE
SALTO-4443: Salesforce configurable aliases for Data Records

### PR DESCRIPTION
**Pull request by tamtamirr**
Added the ability to configure aliases of Data Records.

---

This is similar to how we support **saltoIDSettings**.
As oppose to how we handle invalid id fields, here we only warn the user about invalid fields in his configuration.
This is how the fetch warning looks like for some invalid alias fields:
Screenshot 2023-07-17 at 15 05 21

---
_Release Notes_: 
Salesforce Adapter:
- Added the ability to configure aliases of Data Records.

---
_User Notifications_: 
_None_

### CODE REVIEW

**Review by amitgoldarb — APPROVED** [review]
Looks great!
Nice work here.
Added a single comment, approving to save cycles.

**Review by benisraeladi — APPROVED** [review]
Some small comments, other than that looks good!

File: packages/salesforce-adapter/config_doc.md [hunk 1]
```diff
@@ -162,12 +162,13 @@ salesforce {
...
+|-------------------------------------------------------------------|--------------------------------------------------|-----------------------------------------------------------------------------------------------------------|
+| includeObjects                                                    | N/A (required when dataManagement is configured) | Data records of matched object names will be fetched                                                      |
+| excludeObjects                                                    | []                                               | Data records of matched object names will not be fetched in case they are matched in includeObjects       |
+| allowReferenceTo                                                  | []                                               | Data records of matched object names will be fetched only when referenced from other fetched data records |
+| [saltoIDSettings](#salto-id-settings-configuration-options)       | N/A (required when dataManagement is configured) | Configuration for cross environments data record ids management                                           |
```
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/config_doc.md:170]: reformat

File: packages/salesforce-adapter/cpq-sbaa-config-doc.md [hunk 2]
```diff
@@ -26,13 +26,14 @@ Paste the following configuration snippet under the include fetch metadata secti
 ```
 
 ### Adding Relevant Data Records to the Salto workspace
-Managing salesforce data records within a Salto workspace is disabled by default and its reuqired to add a `data` configuration section in `/salto.config/adapters/salesforce.nacl`
+Managing salesforce data records within a Salto workspace is disabled by default and its required to add a `data` configuration section in `/salto.config/adapters/salesforce.nacl`
```
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/cpq-sbaa-config-doc.md:29]: This change is unrelated to my PR, just fixed my typo. But rephrased

**Pull request merged**